pkg/collect/cluster_resources_test.go

```diff
@@ -1,14 +1,200 @@
+	tests := []struct {
+		name      string
+		leaseName string
+	}{
+		{
+			name:      "single namespace",
+			leaseName: "default",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := testclient.NewSimpleClientset()
+			ctx := context.Background()
+			err := createTestVolumeAttachments(client, tt.leaseName)
+			require.NoError(t, err)
+
+			volumeAttachments, _ := volumeAttachments(ctx, client)
+			assert.NotEmpty(t, volumeAttachments)
+		})
+	}
+}
+
+func createTestVolumeAttachments(client kubernetes.Interface, leaseName string) error {
+	_, err := client.StorageV1().VolumeAttachments().Create(context.Background(), &storagev1.VolumeAttachment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: leaseName,
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func Test_Leases(t *testing.T) {
+	tests := []struct {
+		name       string
+		leaseName  string
+		namespaces []string
+	}{
+		{
+			name:       "single namespace",
+			leaseName:  "default",
+			namespaces: []string{"default"},
+		},
+		{
+			name:       "multiple namespaces",
+			leaseName:  "default",
+			namespaces: []string{"default", "test"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := testclient.NewSimpleClientset()
+			ctx := context.Background()
+			err := createTestLeases(client, tt.leaseName, tt.namespaces)
+			require.NoError(t, err)
+
+			leases, _ := leases(ctx, client, tt.namespaces)
+			assert.Equal(t, len(tt.namespaces), len(leases))
+
+			for _, ns := range tt.namespaces {
+				assert.NotEmpty(t, leases[ns+".json"])
+			}
+		})
+	}
+}
+
+func createTestLeases(client kubernetes.Interface, leaseName string, namespaces []string) error {
+	for _, ns := range namespaces {
+		_, err := client.CoordinationV1().Leases(ns).Create(context.Background(), &v1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: leaseName,
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Test_ServiceAccounts(t *testing.T) {
+	tests := []struct {
+		name               string
+		serviceAccountName string
+		namespaces         []string
+	}{
+		{
+			name:               "single namespace",
+			serviceAccountName: "default",
+			namespaces:         []string{"default"},
+		},
+		{
+			name:               "multiple namespaces",
+			serviceAccountName: "default",
+			namespaces:         []string{"default", "test"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := testclient.NewSimpleClientset()
+			ctx := context.Background()
+			err := createTestServiceAccounts(client, tt.serviceAccountName, tt.namespaces)
+			require.NoError(t, err)
+
+			servicesAccounts, _ := serviceAccounts(ctx, client, tt.namespaces)
+			assert.Equal(t, len(tt.namespaces), len(servicesAccounts))
+
+			for _, ns := range tt.namespaces {
+				assert.NotEmpty(t, servicesAccounts[ns+".json"])
```

Comment:
do we want to check that the service account name exists in the JSON...? at least checking 1 element would give confidence its valid JSON

Reply:
Thank you! I have added json content check to check those elements have been written,